x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -5520,6 +5520,43 @@ public void testMvSortInvalidOrder() {
         assertEquals("Invalid order value in [mv_sort(v, o)], expected one of [ASC, DESC] but got [dsc]", iae.getMessage());
     }
 
+    /**
+     * Tests added for #105383, STATS BY constant.
+     */
+    public void testStatsByConstant() {
+        var plan = plan("""
+            from test
+            | stats m = max(salary), a = avg(salary) by 0
+            | keep m, a
+            """);
+
+        var project = as(plan, Project.class);
+        var eval = as(project.child(), Eval.class);
+        var limit = as(eval.child(), Limit.class);
+        var agg = as(limit.child(), Aggregate.class);
+
+        assertThat(agg.aggregates(), hasSize(3));
+        assertThat(agg.groupings(), hasSize(1));
+        var ref = as(agg.groupings().get(0), ReferenceAttribute.class);
+        assertEquals("0", ref.name());
```

Comment:
IMHO I'm not sure we're testing the right thing here.

I _guess_ what's happening is that our optimizer creates an `EVAL` in front of the `STATS` that evaluates the `0`. That's why we see're seeing a reference attribute here.

But that's not what _needs_ to happen. In fact, since the group is foldable, we could optimize away the `by` clause entirely, and, more cheaply, add an `EVAL` after the `STATS` that just appends the `0` to every row.

In that sense, asserting that we have a reference attribute `0` here might be overconstraining.

Reply:
> IMHO I'm not sure we're testing the right thing here.
> 
> I _guess_ what's happening is that our optimizer creates an `EVAL` in front of the `STATS` that evaluates the `0`. That's why we see're seeing a reference attribute here.
> 
> But that's not what _needs_ to happen. In fact, since the group is foldable, we could optimize away the `by` clause entirely, and, more cheaply, add an `EVAL` after the `STATS` that just appends the `0` to every row.
> 
> In that sense, asserting that we have a reference attribute `0` here might be overconstraining.

Thanks for reviewing @alex-spies ! Let me move the tests to CsvTests, I tried to avoid adding a new EsqlCapability just for adding new tests, but perhaps that's the right thing to do.